skate-plugin/src/main/kotlin/com/slack/sgp/intellij/tracing/SkateTracingEvent.kt

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (C) 2024 Slack Technologies, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.slack.sgp.intellij.tracing
+
+interface SkateTracingEvent {
```

Comment:
For namespacing and sealedness, what if we make this `sealed interface SkateTracingEvent` and nested all the subtypes as nested classes in this without the `Event` suffix? i.e. `SkateTracingEvent.ProjectGen.PROJECT_GEN_OPEN`